source/expressions.tex

```diff
@@ -1280,6 +1279,16 @@
 A \grammarterm{lambda-expression} does not introduce a class scope.
 \end{note}
 
+\pnum
+If the expression \tcode{this}
+appears within the predicate of a contract assertion\iref{basic.contract.general}
+(including as the result of an implicit transformation\iref{expr.prim.id.general}
+and including in the bodies of nested \grammarterm{lambda-expression}s), %JMB: I think this is hard to parse, and lori suggested removing the second comma.  It's "If (X including Y) and Z then A" but we have "If X and Y, and Z, then A".   Should we make this "If X (including Y) and Z, A"
```

Comment:
```suggestion
and including in the bodies of nested \grammarterm{lambda-expression}s)
```

Yes, that comma should go.  We have "If X (including Y) and Z, go home."  No comma before the quoted and.

Reply:
ok, removing the comma.   Can we remove the first and?  I still find it hard because we have "If X and including Y and Z" --- it might be clearer to parse if we had "If X including Y and Z"?

Reply:
I don't know what you mean.  We have parentheses:

"If X (including A and including B) and Y, go home." 

The parentheses group this sufficiently, in my view, and make the first "and" sort-of required.

Reply:
Ignore me, i was somehow (repeatedly) horribly misparsing the whole sentence.  It's good now.